Flaky smart contract test

One of our tests in smart contracts has been failing from time to time.
Examples of failed run https://github.com/unlock-protocol/unlock/actions/runs/3329050274/jobs/5505784564

```
1) Contract: Lock / onKeyExtendHook
       renewMembershipFor
         key cancels should log the hook event:

      AssertionError: expected '17434004751277' to equal '17434004751278'
      + expected - actual

      -17434004751277
      +17434004751278
      
      at Context.<anonymous> (test/Lock/onKeyExtendHook.js:109:14)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at runNextTicks (node:internal/process/task_queues:65:3)
      at listOnTimeout (node:internal/timers:528:9)
      at processTimers (node:internal/timers:502:7)
```

--- a/smart-contracts/test/Lock/onKeyExtendHook.js
+++ b/smart-contracts/test/Lock/onKeyExtendHook.js
@@ -6,6 +6,7 @@ const {
   ADDRESS_ZERO,
   purchaseKey,
   reverts,
+  almostEqual,
 } = require('../helpers')
 
 const keyPrice = ethers.utils.parseUnits('0.01', 'ether')
@@ -106,10 +107,7 @@ contract('Lock / onKeyExtendHook', (accounts) => {
       assert.equal(args.msgSender, lock.address)
       assert.equal(args.tokenId.toString(), tokenId.toString())
       assert.equal(args.from, lockOwner.address)
-      assert.equal(
-        tsBefore.add(expirationDuration).toString(),
-        args.newTimestamp.toString()
-      )
+      assert(almostEqual(tsBefore.add(expirationDuration), args.newTimestamp))
       assert.equal(tsBefore.toString(), args.prevTimestamp.toString())
     })
   })

--- a/smart-contracts/test/helpers/index.js
+++ b/smart-contracts/test/helpers/index.js
@@ -11,6 +11,7 @@ const time = require('./time')
 const interface = require('./interface')
 const uniswapV2 = require('./uniswapV2')
 const uniswapV3 = require('./uniswapV3')
+const math = require('./math')
 
 module.exports = {
   getBalance,
@@ -26,4 +27,5 @@ module.exports = {
   ...interface,
   ...uniswapV2,
   ...uniswapV3,
+  ...math,
 }

--- a/smart-contracts/test/helpers/math.js
+++ b/smart-contracts/test/helpers/math.js
@@ -1,0 +1,5 @@
+const almostEqual = (bn1, bn2, epsilon = 2) => Math.abs(bn1.sub(bn2)) < epsilon
+
+module.exports = {
+  almostEqual,
+}

